### PR TITLE
Add integration test

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

Adds an integration test job to CI.

## What is the intent behind these changes?

So we have as quick feedback as possible.

## Where should the reviewer start?

Please review commit-by-commit.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,46 +3,52 @@ version: 2.1
...
+executors:
+  integration:
+    docker:
+      - image: circleci/node:10-buster-browsers
+      - image: rodolpheche/wiremock:2.27.2-alpine
+        command: ["--port", "9091"]
+      - image: bitnami/redis:6.0
+        environment:
+          ALLOW_EMPTY_PASSWORD: "yes"
+      - image: quay.io/hmpps/hmpps-auth:latest
```
Comment (tomsmyers) [inline .circleci/config.yml:17]: this isn't needed - it's covered by wiremock

Comment (sldblog) [reply .circleci/config.yml]: removed in 275fe55

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,46 +3,21 @@ version: 2.1
...
       - store_artifacts:
           path: 'pact/pacts'
-      - run:
-          command: |
-            npm run build
-            DATE=$(date '+%Y-%m-%d')
-            export BUILD_NUMBER=${DATE}.${CIRCLE_BUILD_NUM}
-            export GIT_REF="$CIRCLE_SHA1"
-            npm run record-build-info
-      - run: # Run linter after build because the integration test code depends on compiled typescript...
```
Comment (lawrence-forooghian) [inline .circleci/config.yml:36]: Could we keep this comment in? I don't actually know _what_ it means but seems like somebody thought it was worth pointing out...

File: .circleci/config.yml [hunk 3]
```diff
@@ -14,11 +14,6 @@ executors:
       - image: bitnami/redis:6.0
         environment:
           ALLOW_EMPTY_PASSWORD: "yes"
-      - image: quay.io/hmpps/hmpps-auth:latest
```
Comment (lawrence-forooghian) [inline .circleci/config.yml]: Can we make this a fixup of the previous commit? I think in general that a change introduced and removed in the same PR should just be wiped from the commit history.

**Review by lawrence-forooghian — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (sldblog) [reply .circleci/config.yml:36]: Done in c45f996: I opted to copy the entire `run lint` step into the integration_test job, following the comment

File: .circleci/config.yml (hunk 3 above)
Comment (sldblog) [reply .circleci/config.yml]: done in c45f996

**Review by lawrence-forooghian — APPROVED** [review]

**Pull request merged**